### PR TITLE
Add option to get all available facet info

### PR DESCRIPTION
**Pull request by chejennifer**
- add option in v2 observations to just get all the available facet info
- use the new ObsTimeSeriesFacet and ObsCollectionFacet caches

### CODE REVIEW

File: internal/server/v2/parser.go [hunk 1]
```diff
@@ -185,3 +190,29 @@ func ParseLinkedNodes(expr string) (*LinkedNodes, error) {
 	}
 	return g, nil
 }
+
+// Gets the contained in place from an expression.
+func GetContainedInPlace(expr string) (*ContainedInPlace, error) {
```
Comment (shifucun) [inline internal/server/v2/parser.go:195]: Rename to `ParseContainedInPlace`? This sounds like a data fetch.

File: internal/server/v2/observation/golden/contained_in_facet_test.go [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+				[]string{"Count_Person", "Median_Age_Person"},
+				"Earth<-containedInPlace+{typeOf:Country}",
+				"LATEST",
+				"country.json",
+			},
+			{
+				[]string{"Count_Person", "Median_Age_Person"},
+				"nuts/FR412<-containedInPlace+{typeOf:AdministrativeArea4}",
+				"",
+				"read_series_bt.json",
```
Comment (shifucun) [inline internal/server/v2/observation/golden/contained_in_facet_test.go:84]: name this to "Meuse_AA4_all.json"?

File: internal/server/v2/observation/facet.go [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+	// Init response
+	result := &pbv2.ObservationResponse{
+		ByVariable: map[string]*pbv2.VariableObservation{},
+		Facets:     map[string]*pb.Facet{},
+	}
+	for _, variable := range variables {
+		result.ByVariable[variable] = &pbv2.VariableObservation{
+			ByEntity: map[string]*pbv2.EntityObservation{},
+		}
+		result.ByVariable[variable].ByEntity[""] = &pbv2.EntityObservation{}
```
Comment (shifucun) [inline internal/server/v2/observation/facet.go:48]: Add a comment on why use ""

File: internal/server/v2/observation/facet.go [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+			}
+			svToFacetList[sv] = append(svToFacetList[sv], row.Data.(*pb.Facets).GetFacets()...)
+		}
+	}
+	// Go through each list of facets, sort and remove duplicates, and add to
+	// result.
+	for sv, facetList := range svToFacetList {
+		entityObservation := &pbv2.EntityObservation{}
+		sort.Sort(ranking.FacetByRank(facetList))
+		seenFacets := map[string]string{}
```
Comment (shifucun) [inline internal/server/v2/observation/facet.go:66]: very nit: golang set uses empty struct like `map[string]struct{}{}` for seenFacets

File: internal/server/v2/observation/facet.go [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+			totalSeries := len(variables) * len(childPlaces)
+			if totalSeries > maxSeries {
+				return nil, status.Errorf(
+					codes.Internal,
+					"Stop processing large number of concurrent observation series: %d",
+					totalSeries,
+				)
+			}
+			log.Println("Fetch series cache in contained-in observation query")
+			btData, err := stat.ReadStatsPb(ctx, store.BtGroup, childPlaces, variables)
```
Comment (shifucun) [inline internal/server/v2/observation/facet.go:140]: should here call SeriesFacet()?

**Review by shifucun — COMMENTED** [review]
I wonder why mixer-grpc.pb has such a big change..

File: internal/server/v2/parser.go (hunk 1 above)
Comment (chejennifer) [reply internal/server/v2/parser.go]: done

File: internal/server/v2/observation/facet.go (hunk 3 above)
Comment (chejennifer) [reply internal/server/v2/observation/facet.go:50]: done

File: internal/server/v2/observation/facet.go (hunk 4 above)
Comment (chejennifer) [reply internal/server/v2/observation/facet.go]: done

File: internal/server/v2/observation/facet.go (hunk 5 above)
Comment (chejennifer) [reply internal/server/v2/observation/facet.go:148]: SeriesFacet won't work if there is a specific query date. Updated to call SeriesFacet if no specified date.

**Comment by chejennifer** [comment]
> I wonder why mixer-grpc.pb has such a big change..

I'm not sure .. are there things I should check?

File: deploy/storage/base_bigtable_info.yaml [hunk 6]
```diff
@@ -4,7 +4,7 @@ tables:
   - auto2w_2023_07_25_10_35_37
   - auto1d_2023_07_31_21_08_49
   - auto1w_2023_07_30_21_04_03
-  - frequent_2023_07_31_20_05_22
+  - frequent_2023_08_10_20_03_02
```
Comment (ajaits) [inline deploy/storage/base_bigtable_info.yaml:7]: Can this be bumped to the latest: frequent_2023_08_18_02_22_08 to get the recent worldbank import?

Comment (chejennifer) [reply deploy/storage/base_bigtable_info.yaml]: done

**Review by shifucun — APPROVED** [review]
thanks for the update!

**Review by ajaits — APPROVED** [review]

**Pull request merged**